java-manta-client/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -1431,24 +1431,21 @@ public void putDirectory(final String rawPath, final boolean recursive,
             return;
         }
 
-        final String[] parts = rawPath.split(SEPARATOR);
-        final Iterator<Path> itr = Paths.get("", parts).iterator();
-        final StringBuilder sb = new StringBuilder(SEPARATOR);
-
-        for (int i = 0; itr.hasNext(); i++) {
-            final String part = itr.next().toString();
-            sb.append(part);
+        final Integer skipDepth = config.getSkipDirectoryDepth();
+        final RecursiveDirectoryCreationStrategy directoryCreationStrategy;
```

Comment:
We select a directoryCreationStrategy each time putDirectory is called so that the configuration parameter can be meaningfully tuned from the ConfigContextMBean in JMX.

(duplicated for clarity, was collapsed by a pushed commit)